lib/elixir/lib/kernel.ex

```diff
@@ -559,7 +559,7 @@ defmodule Kernel do
 
   Allowed in guard tests. Inlined by the compiler.
   """
-  @spec node(pid | reference | port) :: node
+  @spec node(pid | reference | port) :: node | :nonode@nohost
```

Comment:
I thought being explicit was good.
I have seen a few times other specs in the project where the return is `term | something_else`, `any | something_else` and `atom | nil`
here's a list:
https://gist.github.com/eksperimental/ba212be62d35b6e62d7876c3ce879e04